File: Data Structures/Stack.java

public class Stack {

	private int[] data;
	private int tos = -1;// top of stackk 

	public Stack(int capacity) {
		this.data = new int[capacity];
	}

	public Stack()// contructor chaining
	{
		this(5);
	}

	public int top()// tells element at tos
	{
		if (this.tos == -1) {
			System.out.println("Stack UnderFlow");
			return -1;
		}
		int value = this.data[this.tos]; 
		return value;
	}

	public int pop()// element to delete
	{
		// check for underflow
		if (this.tos == -1) {
			System.out.println("Stack UnderFlow");
			return -1;
		}
		int value = this.data[this.tos];
		this.data[this.tos] = 0;
		this.tos--;
		return value;

	}

	public void push(int item) {
		// check for overflow
		if (this.tos == this.data.length - 1) {
			System.out.println("Stack OverFlow");
			return;
		}
		this.tos++;
		this.data[this.tos] = item;
	}

	public boolean isEmpty() {
		if (this.size() == 0)
			return true;
		else
			return false;
	}

	public void display() {
		for (int i = tos; i >= 0; i--) {
			System.out.print(this.data[i] + " "); 
		}
		System.out.println();
	}

	public int size() {
		return this.tos + 1;
	}

}

File: Data Structures/Stack/C++/stack.cpp
#include<iostream>
#include<stdlib.h>

using namespace std;

struct node 
{
	int info;
	struct node *link;
}*top=NULL;

void push(int item);
int pop();
int isEmpty();
void display();

int main()
{
	int choice,item,n,i;
	
	while(1)
	{
	
	   cout<<"\n 1. Press 1 to push the elements in the stack";
	   cout<<"\n 2. Press 2 to pop the elements in the stack";
       cout<<"\n 3. Press 3 to display all the elements in the stack";
       cout<<"\n 4. Press 4 to quit";
    
	   cout<<"\n Enter your choice";
	   cin>>choice;
	   
	   switch(choice)
	   {
	   	case 1:
	   		cout<<"\n Enter the number of items you want to push";
	   		cin>>n;
	   		cout<<"\n Enter the items :";
	   		
	   		for(i=0;i<n;i++)
	   		{
			cin>>item;   
	   		push(item);
	   	    }  
			break;
	   		
	   	case 2:
		   	cout<<"\n Enter the number of items you want to pop";
	   		cin>>n;
	   		for(i=0;i<n;i++)
	   		{
	   			int popped_number;
	   			popped_number=pop();
	   			cout<<"\n Popped item is :"<<popped_number;
			   }
			break;
			
		case 3:
		     display();
			 break;
			 
		case 4:
		    exit(1);
		
		default:
		cout<<"\n Enter a valid choice";
		
		}/*End of switch*/
	
	
	} /*End of while*/
	
	
}/*End of main*/

void push(int item)
{
	struct node *tmp;
	tmp=(struct node *)malloc(sizeof(struct node));
	
	if(tmp==NULL)
	{
		cout<<"\n Stack overflow";
		return ;
	}
	
    tmp->info=item;
    tmp->link=top;
    top=tmp;
}/* End of push*/

int pop()
{
	struct node *tmp;
	int item;
	if(isEmpty())
	{
		cout<<"\n Stack underflow";
		exit(1);
		
	}
	tmp=top;
	item=tmp->info;
	top=top->link;
	free(tmp);
	return item;
}/*End of pop*/

int isEmpty()
{
	if(top==NULL)
	return 1;
	
	else 
	return 0;
}/* End of isEmpyu*/

void display()
{
	struct node *ptr;
	ptr=top;
	
	if(isEmpty())
	{
		cout<<"\n Stack is empty";
		return;
	}
	
	cout<<"\n Stack elements:\n";
	
	while(ptr!=NULL)
	{
		cout<<"\n"<<ptr->info;
		ptr=ptr->link;
	}
	cout<<endl;
	
}/*End of display*/
